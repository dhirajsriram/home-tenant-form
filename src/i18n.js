import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { initReactI18next } from "react-i18next";

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    // we init with resources
    resources: {
      de: {
        translations: {
          "Name": "Name",
          "Email":"Email",
          "Phone": "Phone",
          "Salary": "Gehalt",
          "Review": "Rezension",
          "Enter your name": "Gib deinen Namen ein",
          "First Name": "Vorname",
          "First name": "Vorname",
          "Last Name": "Nachname",
          "Last name": "Nachname",
          "Enter your Email Address": "Geben sie ihre E-Mail Adresse ein",
          "Enter your phone": "Geben Sie Ihr Telefon ein",
          "Enter your Salary": "Geben Sie Ihr Gehalt ein",
          "Kindly review your input": "Bitte überprüfen Sie Ihre Eingabe",
          "Over 4000": "Mehs als 4000",
          "submit":"Registrieren",
          next: "weiter",
          back: "Bisherige"
        },
      },
      en: {
        translations: {
          "Name": "Name",
          "Email":"Email",
          "Phone": "Phone",
          "Salary": "Salary",
          "Review": "Review", 
          "Enter your name": "Enter your name",
          "First Name": "First Name",
          "First name": "First Name",
          "Last Name": "Last Name",
          "Last name": "Last Name",
          "Enter your Email Address": "Enter your Email Address",
          "Enter your phone": "Enter your Email Phone",
          "Enter your Salary": "Enter your Email Salary",
          "Kindly review your input":"Kindly review your input",
          "Over 4000": "Over 4000",
          "submit":"submit",
          next: "next",
          back: "back"
        },
      },
    },
    fallbackLng: "en",
    debug: true,

    // have a common namespace used around the full app
    ns: ["translations"],
    defaultNS: "translations",

    keySeparator: false, // we use content as keys

    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
