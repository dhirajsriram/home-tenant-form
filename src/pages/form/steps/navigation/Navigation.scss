.form-button-container {
    padding-top: 50px;
    display: flex;
    justify-content: space-between;
}