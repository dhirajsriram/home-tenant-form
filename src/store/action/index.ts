import { SETLANGUAGE, SETSTEP, SETFORMVALUE, SETINITIAL } from "../constants/constants";

interface Actions {
  type: string;
  payload: any;
}

export const setStep = (step: number): Actions => {
  return {
    type: SETSTEP,
    payload: { step },
  };
};

export const setFormValue = (payload: any): Actions => {
  return {
    type: SETFORMVALUE,
    payload: payload,
  };
};

export const setInitial = (): Actions => {
  return {
	type: SETINITIAL,
	payload: {}
  };
};

export const setLanguage = (language: string): Actions => {
  return {
    type: SETLANGUAGE,
    payload: { language },
  };
};
