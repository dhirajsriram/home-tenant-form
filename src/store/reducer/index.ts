import renameReducer from "./rename";
import { combineReducers } from "redux"

const allReducers =  combineReducers({
	renameReducer
});

export default allReducers
