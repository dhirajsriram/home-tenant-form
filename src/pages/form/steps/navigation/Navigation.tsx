import React from "react";
import { Button } from "@material-ui/core";
import { useDispatch, useSelector } from "react-redux";
import { SETSTEP } from "../../../../store/constants/constants";
import { useTranslation } from "react-i18next";
import "./Navigation.scss";

interface navigation {
  next: number;
  previous: number;
  field: string;
}

const Navigation: React.FC<navigation> = (props) => {
  const dispatch = useDispatch();
  const { t } = useTranslation();
  const form: any = useSelector((state: any) => state.reducer.form);

  const next = (e: React.MouseEvent): void => {
    // Move forward 
    dispatch({ type: SETSTEP, payload: { step: props.next } });
  };

  const back = (e: React.MouseEvent): void => {
    // Move back 
    dispatch({ type: SETSTEP, payload: { step: props.previous } });
  };

  const valid = (field: string): boolean => {
    // Check if the field are valiid
    if (field) {
      if (field === "name") {
        return !(form["first-name"].valid && form["last-name"].valid);
      }
      return !form[props.field].valid;
    }
    return false;
  };

  return (
    <div className="form-button-container">
      <Button
        variant="contained"
        color="primary"
        className="form-button-next"
        onClick={(e) => next(e)}
        disabled={valid(props.field)}
      >
        {props.next === 5 ? t("submit") : t("next")}
      </Button>
      {props.field !== "name" && (
        <Button
          className="form-button-previous"
          color="primary"
          onClick={(e) => back(e)}
        >
          {t("back")}
        </Button>
      )}
    </div>
  );
};

export default Navigation;
