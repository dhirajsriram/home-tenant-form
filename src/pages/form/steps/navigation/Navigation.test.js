import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure, mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { Provider } from "react-redux";
import { createStore } from "redux";
import allReducers from "../../../../store/reducer";
import Navigation from "./Navigation";
import "../../../../i18n";
import { setFormValue } from "../../../../store/action";

let wrapper;
let props = {
  next: 4,
  previous: 2,
  field: "email",
};
configure({ adapter: new Adapter() });

let store = createStore(allReducers);

beforeEach(() => {
  wrapper = mount(
    <Provider store={store}>
      <React.Suspense fallback={<div></div>}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Navigation {...props} />
        </MemoryRouter>
      </React.Suspense>
    </Provider>
  );
});

describe("<Navigation>", () => {
  it("Renders without exploding", async () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Goes back", async () => {
    const button = wrapper.find("button").at(1);
    button.simulate("click");
    expect(store.getState().reducer.step).toBe(props.previous);
  });
  it("Next is disabled by default", async () => {
    const button = wrapper.find("button").at(0);
    expect(button.props().disabled).toBe(true);
  });
  it("Goes forward", async () => {
    let mockFromValue = {
      field: "email",
      value: {
        value: "XXXX",
        valid: true,
      },
    };
    store.dispatch(setFormValue(mockFromValue));
    const button = wrapper.find("button").at(0);
    button.simulate("click");
    expect(store.getState().reducer.step).toBe(props.next);
  });
});
