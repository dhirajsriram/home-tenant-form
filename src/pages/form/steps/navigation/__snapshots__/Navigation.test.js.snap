// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation> Renders without exploding 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Suspense
    fallback={<div />}
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/",
        ]
      }
      keyLength={0}
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Navigation
          field="email"
          next={4}
          previous={2}
        >
          <div
            className="form-button-container"
          >
            <WithStyles(ForwardRef(Button))
              className="form-button-next"
              color="primary"
              disabled={true}
              onClick={[Function]}
              variant="contained"
            >
              <ForwardRef(Button)
                className="form-button-next"
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                    "disableElevation": "MuiButton-disableElevation",
                    "disabled": "Mui-disabled",
                    "endIcon": "MuiButton-endIcon",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "startIcon": "MuiButton-startIcon",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                    "textSizeLarge": "MuiButton-textSizeLarge",
                    "textSizeSmall": "MuiButton-textSizeSmall",
                  }
                }
                color="primary"
                disabled={true}
                onClick={[Function]}
                variant="contained"
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-contained form-button-next MuiButton-containedPrimary Mui-disabled"
                  component="button"
                  disabled={true}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-contained form-button-next MuiButton-containedPrimary Mui-disabled"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={true}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained form-button-next MuiButton-containedPrimary Mui-disabled Mui-disabled"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        next
                      </span>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
            <WithStyles(ForwardRef(Button))
              className="form-button-previous"
              color="primary"
              onClick={[Function]}
            >
              <ForwardRef(Button)
                className="form-button-previous"
                classes={
                  Object {
                    "colorInherit": "MuiButton-colorInherit",
                    "contained": "MuiButton-contained",
                    "containedPrimary": "MuiButton-containedPrimary",
                    "containedSecondary": "MuiButton-containedSecondary",
                    "containedSizeLarge": "MuiButton-containedSizeLarge",
                    "containedSizeSmall": "MuiButton-containedSizeSmall",
                    "disableElevation": "MuiButton-disableElevation",
                    "disabled": "Mui-disabled",
                    "endIcon": "MuiButton-endIcon",
                    "focusVisible": "Mui-focusVisible",
                    "fullWidth": "MuiButton-fullWidth",
                    "iconSizeLarge": "MuiButton-iconSizeLarge",
                    "iconSizeMedium": "MuiButton-iconSizeMedium",
                    "iconSizeSmall": "MuiButton-iconSizeSmall",
                    "label": "MuiButton-label",
                    "outlined": "MuiButton-outlined",
                    "outlinedPrimary": "MuiButton-outlinedPrimary",
                    "outlinedSecondary": "MuiButton-outlinedSecondary",
                    "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                    "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                    "root": "MuiButton-root",
                    "sizeLarge": "MuiButton-sizeLarge",
                    "sizeSmall": "MuiButton-sizeSmall",
                    "startIcon": "MuiButton-startIcon",
                    "text": "MuiButton-text",
                    "textPrimary": "MuiButton-textPrimary",
                    "textSecondary": "MuiButton-textSecondary",
                    "textSizeLarge": "MuiButton-textSizeLarge",
                    "textSizeSmall": "MuiButton-textSizeSmall",
                  }
                }
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiButton-root MuiButton-text form-button-previous MuiButton-textPrimary"
                  component="button"
                  disabled={false}
                  focusRipple={true}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                  type="button"
                >
                  <ForwardRef(ButtonBase)
                    className="MuiButton-root MuiButton-text form-button-previous MuiButton-textPrimary"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-text form-button-previous MuiButton-textPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        back
                      </span>
                      <WithStyles(memo)
                        center={false}
                      >
                        <ForwardRef(TouchRipple)
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(Button)>
            </WithStyles(ForwardRef(Button))>
          </div>
        </Navigation>
      </Router>
    </MemoryRouter>
  </Suspense>
</Provider>
`;
